compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2713,15 +2713,26 @@ public void visit(BLangInvocation iExpr) {
     public void visit(BLangErrorConstructorExpr errorConstructorExpr) {
         String typeName = "";
         String pkgAlias = "";
-        if (errorConstructorExpr.errorTypeRef == null) {
+        BSymbol symbol = null;
+
+        if (errorConstructorExpr.errorTypeRef == null && types.isAssignable(expType, symTable.errorType)) {
```

Comment:
Can't we adopt the same approach as the one we've used when type checking mapping and list constructors against unions? That way we can handle unions of errors and non-errors as the CET type too.

https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L1601